Added class DistributedNetworkAction

kummy98 opened the pull request:
The class is not finished yet ,the last 3 methods are not implemented

turcato-niccolo committed:
-Added abstract class NodeActionStructure<T>

turcato-niccolo committed:
-Updated abstract class NodeActionStructure<T>

Commit:
added class DistributedNetworkAction
Some method still need to be implemented (setDestination getDestination and getSender)

turcato-niccolo commented on distributednetworkmanager/src/main/java/com/gruppo1/distributednetworkmanager/DistributedNetworkAction.java:
```diff
@@ -0,0 +1,305 @@
+                        && extra.equals(DEFAULT_IGNORED) && payload.equals(DEFAULT_IGNORED)&& !param.equals(DEFAULT_IGNORED))
+
+                || ((actionCommand==Type.FIND_VALUE || actionCommand==Type.ANSWER_FIND_VALUE ) &&
+                        payload.equals(DEFAULT_IGNORED) && !extra.equals(DEFAULT_IGNORED) &&!extra.equals(DEFAULT_IGNORED))
```
Checking extra two times

kummy98 replied:
Yeah one of that was !param.equals(DEFAULT_IGNORED), i'll change it now

ricdezen commented on distributednetworkmanager/src/main/java/com/gruppo1/distributednetworkmanager/DistributedNetworkAction.java:
```diff
@@ -0,0 +1,305 @@
+    }
+
+    public DistributedNetworkAction setNodeId(int nodeId)
+    {
```
Are you sure you need to set fields like this in this first 3 methods? Just setting them returning void should be enough

turcato-niccolo commented on distributednetworkmanager/src/main/java/com/gruppo1/distributednetworkmanager/DistributedNetworkAction.java:
```diff
@@ -0,0 +1,305 @@
+        static final int PING = 0;
+        //<#>[ID] [1/1] <#>ANSWER_PING [NODE_ID] [IGNORED] [IGNORED]
+        static final int ANSWER_PING = 1;
+        //<#>[ID] [k/N] STORE [NODE_ID] [IGNORED] [VALUE]
```
Qui è
[ID] [k/N] STORE [NODE_ID] [NAME] [VALUE]
Poichè chiedi di memorizzare una risorsa

RicDeZen commented:
I don't see any big problems currently but I'd have to look deeper

turcato-niccolo commented on distributednetworkmanager/src/main/java/com/gruppo1/distributednetworkmanager/DistributedNetworkAction.java:
```diff
@@ -0,0 +1,305 @@
+        static final int ANSWER_FIND_NODE = 5;
+        //<#>[ID] [k/N] FIND_VALUE [NODE_ID] [YES_RESOURCE] [IGNORED]
+        static final int FIND_VALUE = 6;
+        //<#>[ID] [k/N] ANSWER_FIND_VALUE [NODE_ID] [YES_RESOURCE] [VALUE]
```
Same as before [ID] [k/N] ANSWER_FIND_VALUE [NODE_ID] [NAME] [VALUE] 
Since a peer asks for a resource value but might only know the hash of its name

RicDeZen commented:
Ideally I'd change maybe the names of the classes since they are a bit long, I think NetworkAction for the abstract class and KadAction for the implementation could be ok

turcato-niccolo commented:
> Ideally I'd change maybe the names of the classes since they are a bit long, I think NetworkAction for the abstract class and KadAction for the implementation could be ok

NetworkAction was already used in the replicated network if I recall correctly xD

ricdezen commented on distributednetworkmanager/src/main/java/com/gruppo1/distributednetworkmanager/DistributedNetworkAction.java:
```diff
@@ -0,0 +1,305 @@
+        return this;
+    }
+
+    public DistributedNetworkAction(int actionCommand, @NonNull String param, @NonNull String extra,@NonNull String payload){
```
Although I don't like having lots of parameters, it would be better to add current and totalMessage to the constructor.
A Builder or Factory pattern can help with the number of parameters but that's for when everything else works

kummy98 replied:
I had everything on the constructor before the builder pattern😂😅.
I'll change it again then

ricdezen replied:
Well this is not really a Builder pattern, there would be a class dedicated to building the Action, but then the constructor of the main class would be hidden. Since you've put some default values which I didn't notice you can keep it like it is for now

Commit:
-Changed STORE's extra from IGNORED to NAME
-Changed ANSWER_FIND_VALUE's extra from YES_RESOURCES to NAME
-Added totalMassages and current messages to the constructor
-removed the 2 extra check

Commit:
-Removed totalMassages and current messages from the constructor
-added 2 methods to set currentMessage and totalMessage

Commit:
-changed return object from Node to SMSPeer in methods getDestination,
getSender and the parameter of sedDestination from Node to SMSPeer

ricdezen reviewed (approved):
~~I kinda need this so I don't really have the time to be picky~~
The hell am I talking about I can simply pull I honestly need to take a break
